contrib/upgrade.sh

```diff
@@ -10,6 +10,17 @@
 #   upgrade.sh 1.15.10
 #   giteahome=/opt/gitea giteaconf=$giteahome/app.ini upgrade.sh
 
+
+
+# Check if gitea service is running
+if ! systemctl is-active gitea &> /dev/null; then
+  echo "Error: gitea is not running."
+  exit 1
+fi
+
+# Continue with rest of the script if gitea is running
+echo "Gitea is running. Continuing with rest of script..."
```

Comment:
Unfortunately, it's not that easy.
As you can see a few lines below, there is no guarantee that a user has or uses `systemctl` as they can set themselves what they use.

Reply:
Yup, the `service_start` command below is optional.

There should a configurable `service_is_active` command like `service_start` or `service_status`.

ps: could the exit code of `systemctl status` be used directly?

Reply:
`pidof gitea` maybe as a more standalone check?

Reply:
Sir can I do it like this?------
if ! pidof gitea &> /dev/null; then
  echo "Error: Gitea is not running."
  exit 1
fi

# Continue with rest of the script if Gitea is running
echo "Gitea is running. Continuing with rest of script..."

Reply:
Sir can I do it like this -----

Check if gitea service is running

if ! pidof gitea &> /dev/null; then
  echo "Error: Gitea is not running."
  exit 1
fi

 Continue with rest of the script if Gitea is running

echo "Gitea is running. Continuing with rest of script..."

Reply:
Reading the comment header of this file, I think systemd can actually be assumed:

> his is an update script for gitea installed via the binary distribution
> from dl.gitea.io on linux as systemd service

So I guess this method is fine, @delvh what do you think?

Reply:
Hmm.
The rest of this file is independent of systemd, so I don't think it's a good idea to throw away that independence.
That will probably make exactly those people angry that don't use systemd…

Reply:
Ok, then `pidof gitea` it is.

Reply:
Yeah, I agree re: pidof, as even though systemctl is used within this file, you can still override those commands, especially if platform is openwrt then `/etc/init.d/gitea` will be used.